requirements-common.txt

```diff
@@ -1,7 +1,4 @@
 keras-core
-# Consider handling GPU here.
-torch>=2.0.1+cpu
```

Comment:
Curious what your take is here -- this can definitely be handled differently for the docker configs, but the way I see it these aren't really requirements for KerasNLP. wdyt?

Reply:
My take is basically what is laid out [here](https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/#id6). `install_requires` should list the minimum needed for a valid install, but `requirements.txt` should be an exhaustive and repeatable recipe for a complete environment.

After the next tf release, I think we can get by with just a `requirements.txt` and `requirements-cuda.txt`. These should be used by all our tooling (no listing extra deps in docker files, etc), and used by contributors to set up a development environment that matches our CI.

Short term, we could consider a `requirements-jax.txt`, `requirements-tf.txt` etc, that match our docker setups. Then all our version pinning, hacking, etc, is consolidated to requirements files.

No need to figure this out all on this PR though. Fine to land as is and keep tweaking.